heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java

```diff
@@ -173,4 +174,47 @@ public static double increaseBy(double value, int paddingPercentage) {
     return value + (paddingPercentage * value) / 100;
   }
 
+  /**
+   * Allocate a new container of a given capacity
+   *
+   * @return the number of containers
+   */
+  public static int allocateNewContainer(ArrayList<Container> containers, Resource capacity,
+                                         int paddingPercentage) {
+    containers.add(new Container(capacity, paddingPercentage));
+    return containers.size();
+  }
+
+  /**
+   * Identifies which components need to be scaled down
+   *
+   * @return Map &lt; component name, scale down factor &gt;
+   */
+  public static Map<String, Integer> getComponentsToScaleDown(Map<String,
+      Integer> componentChanges) {
+    Map<String, Integer> componentsToScaleDown = new HashMap<String, Integer>();
+    for (Map.Entry<String, Integer> entry : componentChanges.entrySet()) {
```

Comment:
can we iterate over the keySet() here instead? The naming helps it tend to be more descriptive than using entry.getKey() and entry.getValue().